rest/blip_api_test.go

```diff
@@ -4492,3 +4492,43 @@ func TestBlipLegacyAttachDocUpdate(t *testing.T) {
 		require.Equal(t, keyValueErr.DocumentID, v2Key)
 	}
 }
+
+func TestUnsubChanged(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyAll)
+	rt := NewRestTester(t, &RestTesterConfig{guestEnabled: true})
+	defer rt.Close()
+
+	btc, err := NewBlipTesterClientOptsWithRT(t, rt, nil)
+	require.NoError(t, err)
+	defer btc.Close()
+
+	// Sub changes
+	err = btc.StartPull()
+	require.NoError(t, err)
+	resp := rt.updateDoc("doc1", "", `{"key":"val1"}`)
+	_, found := btc.WaitForRev("doc1", resp.Rev)
+	require.True(t, found)
+
+	// Unsub changes
+	unsubChangesRequest := blip.NewRequest()
+	unsubChangesRequest.SetProfile(db.MessageUnsubChanges)
+	err = btc.pullReplication.sendMsg(unsubChangesRequest)
+	require.NoError(t, err)
+	unsubChangesResponse := unsubChangesRequest.Response()
+	_, err = unsubChangesResponse.Body()
+	require.NoError(t, err)
+
+	// Confirm no more changes being sent
+	resp = rt.updateDoc("doc2", "", `{"key":"val1"}`)
```

Comment:
This test might be racy given it's not a synchronous operation to call `unsubChanges`. If the changes feed takes a while to actually exit, doc2 might end up being sent.

I'm not exactly sure of the fix here - the replication protocol wording implies unsub is async, but it's possible we can block inside the test on stats instead to ensure timing.